detection_rules/devtools.py

```diff
@@ -354,3 +356,217 @@ def rule_survey(ctx: click.Context, query, date_range, dump_file, hide_zero_coun
+
+    if not existing_model_name:
+        click.secho(f'[+] validated no existing models matched names for: '
+                    f'{loaded_contents["model"]["filename"]}', fg='green')
+
+    # save zip
+    zip_name_no_ext = Path(directory).resolve()
+    zip_name = f'{zip_name_no_ext}.zip'
+    shutil.make_archive(str(zip_name_no_ext), 'zip', root_dir=zip_name_no_ext.parent, base_dir=zip_name_no_ext.name)
+    click.secho(f'[+] zipped folder saved to {zip_name} for release', fg='green')
+
+    click.secho(f'[!] run `setup-dga-model -d {directory}` to test this on a live stack before releasing!', fg='yellow')
+
+    description = {
+        'model_name': model_name + '\n\n----\n\n',
+        'date': now,
+        'model_sha256': model_hash
+    }
+    description_str = '\n'.join([f'{k}: {v}' for k, v in description.items()])
+    click.echo()
+    click.echo(f'[*] description to paste with release:\n\n{description_str}\n')
+
+    return zip_name, description_str
+
+
+@gh_release_group.command('validate-ml-detections-asset')
+@click.argument('directory', type=click.Path(exists=True, file_okay=False))
+def validate_ml_detections_asset(directory):
+    """Validate and prep ML detection rules and jobs before release."""
+    import pytoml
+
+    now = time.strftime('%Y-%m-%dT%H:%M:%SZ', time.gmtime())
+
+    job_paths = list(Path(directory).glob('*.json'))
+    rule_paths = list(Path(directory).glob('*.toml'))
+    job_count = len(job_paths)
+    rule_count = len(rule_paths)
+
+    for job in job_paths:
+        try:
+            with open(job, 'r') as f:
+                j = json.load(f)
+                assert j.get('name'), click.style(f'[!] job file "{job}" missing: name', fg='red')
+                assert j.get('type'), click.style(f'[!] job file "{job}" missing: type', fg='red')
+                assert j.get('body'), click.style(f'[!] job file "{job}" missing: body', fg='red')
+        except json.JSONDecodeError as e:
+            client_error(f'Invalid JSON in {job} file', e)
+
+    click.secho(f'[*] validated json formatting and required fields in {job_count} job files', fg='green')
+
+    for rule in rule_paths:
+        with open(rule, 'r') as f:
+            try:
+                pytoml.load(f)
+            except pytoml.TomlError as e:
+                client_error(f'[!] invalid rule toml for: {rule}', e)
+
+    click.secho(f'[*] validated toml formatting for {rule_count} rule files', fg='green')
+
+    # save zip
+    zip_name_no_ext = Path(directory).resolve()
+    zip_name = f'{zip_name_no_ext}.zip'
+    shutil.make_archive(str(zip_name_no_ext), 'zip', root_dir=zip_name_no_ext.parent, base_dir=zip_name_no_ext.name)
+    click.secho(f'[+] zipped folder saved to {zip_name} for release', fg='green')
+
+    click.secho('[!] run `kibana upload-rule` to test rules on a live stack before releasing!', fg='green')
+    click.secho('[!] run `es upload-ml-job` to test jobs on a live stack before releasing!', fg='green')
```

Comment:
we don't need to double the `!`
```suggestion
    click.secho(f'[+] zipped folder saved to {zip_name} for release', fg='green')

    click.secho('[!] run `kibana upload-rule` to test rules on a live stack before releasing', fg='green')
    click.secho('[!] run `es upload-ml-job` to test jobs on a live stack before releasing', fg='green')
```